### ISSUE TITLE
Paths are not getting changed at all

### ISSUE THREAD

**Issue by ncls-alien** [aleclarson/vite-tsconfig-paths#121]
I tried for over 3 hours to find what I am missing but I can't find any mistake, however, none of my paths is being changed by the plugin.
Debug is not outputting anything useful despite the fact that it loads the configs correctly.

Error:
```
Failed to resolve import "@components/titlebar" from "src\App.tsx". Does the file exist?
20:00:56 [vite] Internal server error: Failed to resolve import "@components/titlebar" from "src\App.tsx". Does the file exist?
  Plugin: vite:import-analysis
  File: G:/Eigene Dateien/Documents/Node/electron/pigeon/src/App.tsx:2:21
  16 |  }
  17 |  import "./App.css";
  18 |  import TitleBar from "@components/titlebar";
     |                        ^
  19 |  function App() {
  20 |    return /* @__PURE__ */ jsxDEV("div", { className: "bg-primary text-primary", children: /* @__PURE__ */ jsxDEV(TitleBar, {}, void 0, false, {
```

vite.config.ts:
```TS
import { defineConfig } from 'vite';
import electron from 'vite-plugin-electron';
import renderer from 'vite-plugin-electron-renderer';
import react from '@vitejs/plugin-react';
import tsConfigPaths from 'vite-tsconfig-paths';

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [
    react(),
    tsConfigPaths({
      loose: true
    }),
    electron([
      {
        // Main-Process entry file of the Electron App.
        entry: 'electron/main.ts',
      },
      {
        entry: 'electron/preload.ts',
        onstart(options) {
          // Notify the Renderer-Process to reload the page when the Preload-Scripts build is complete, 
          // instead of restarting the entire Electron App.
          options.reload();
        },
      },
    ]),
    renderer(),
  ],
});
```
tsconfig.json:
```JSON
{
  "compilerOptions": {
    "target": "ES2020",
    "useDefineForClassFields": true,
    "lib": ["ES2020", "DOM", "DOM.Iterable"],
    "module": "ESNext",
    "skipLibCheck": true,
    "baseUrl": "./",

    /* Bundler mode */
    "moduleResolution": "bundler",
    "allowImportingTsExtensions": true,
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "react-jsx",
    "allowJs": false,

    /* Linting */
    "strict": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noFallthroughCasesInSwitch": true,
    "paths": {
      "@/*": [ "src/*" ],
      "@components/*": [ "src/components/*" ],
      "@public/*": [ "public/*" ],
    },
  },
  "include": ["src", "electron"],
  "references": [{ "path": "./tsconfig.node.json" }]
}
```

**Comment by fvsch** [comment]
Couple things to maybe check:

1. Do you need to turn on the `allowJs` option in your tsconfig? This plugin seems to use it as a signal for allowing `.svelte` and `.vue` files, so maybe some of that logic applies to other formats such as `.svg`? https://github.com/aleclarson/vite-tsconfig-paths#allowjs

2. Should your `public` folder be part of your `include` option in the tsconfig? Since you're importing a SVG file from there and treating it as a JS module in your dependency tree, and this plugin's README says it respects include/exclude rules: https://github.com/aleclarson/vite-tsconfig-paths#includeexclude

**Comment by fvsch** [comment]
Also wondering if importing stuff from the `public` directory is a good idea, given that by default Vite treats this directory as content to copy over as-is. There might be some logic on the Vite side that prevents importing from that directory? Can you check with `publicDir: false`?

https://vitejs.dev/config/shared-options.html#publicdir

**Comment by aleclarson** [comment]
This might be a Windows issue

**Comment by aleclarson** [comment]
This might be a Windows issue. Run Vite with `DEBUG="vite-tsconfig-paths vite:resolve-details" vite` and paste the logs here please.

**Comment by aleclarson** [comment]
Oh my bad, you need to do `DEBUG="vite-tsconfig-paths,vite:resolve-details" vite` (note the comma)

**Comment by Stel000** [comment]
Any progress? Same issue with vite-plugin-electron XD

**Comment by webpro** [comment]
Having the same issue here. When checking the debug logs, it looks like exactly the items that match `tsconfig.json#compilerOptions.paths` are marked as `[fallthrough]`, e.g.

```
  vite:resolve-details [fallthrough] @/pages/index +0ms
  vite:resolve-details [fallthrough] @/pages/index +0ms
```

And eventually the same imports are not resolved/found:

```
Failed to load url @/pages/index (resolved id: @/pages/index)
```

Seeing the same in the other debug log above.

The `fallthrough` logic I find at https://github.com/vitejs/vite/blob/eef4aaa063ed420c213cb9e24f680230cf2132b2/packages/vite/src/node/plugins/resolve.ts#L430

Then I found the `getIncluder` logic here in `vite-tsconfig-paths` and I'm convinced that in my case my `include` paths caused the issue.

In my local `lib/tsconfig.json`:

```json
{
  "extends": "./some/path/far/away/tsconfig-base.json",
  "compilerOptions": {
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
```

And that `tsconfig-base.json` contains this:

```json
{
  "include": ["../../../src/**/*.ts"]
}
```

I believe the issue is that (`vite` or) this plugin does not resolve the `include` entries against the file in which they're defined. Because they seem unchanged in the debug logs.

When I override the `include` entries in `lib/tsconfig.json` all is fine. Hope this helps :)

**Comment by Aoshisen** [comment]
I changed `tsconfig.json`  "allowJs" filed value to "true"
![image](https://github.com/aleclarson/vite-tsconfig-paths/assets/66982800/7a4d1e73-a7f0-4e00-a69c-c43d8ab05182)
then the plugin works well

**Comment by Razunter** [comment]
In my case it works with pnpm and doesn't work with npm\bun

`npm run build` log:
```
vite v5.2.11 building for production...
  vite:resolve-details [fs] w:\web\santehnica.ru\bun-issue\index.html -> W:/web/santehnica.ru/bun-issue/index.html +74ms
transforming (1) index.html  vite:resolve-details [url] /src/main.ts -> W:/web/santehnica.ru/bun-issue/src/main.ts +21ms
  vite:resolve-details [relative] ./style.css -> W:/web/santehnica.ru/bun-issue/src/style.css +11ms
  vite:resolve-details [relative] ./typescript.svg -> W:/web/santehnica.ru/bun-issue/src/typescript.svg +1ms
  vite:resolve-details [fallthrough] /vite.svg +0ms
  vite:resolve-details [relative] ./counter.ts -> W:/web/santehnica.ru/bun-issue/src/counter.ts +1ms
  vite:resolve-details [fallthrough] @js/test.ts +0ms
  vite:resolve-details [fallthrough] @js/test.ts +1ms
✓ 3 modules transformed.
x Build failed in 42ms
```

`pnpm run build` log:
```
vite v5.2.11 building for production...
  vite:resolve-details [fs] D:\temp\bun-issue\index.html -> D:/temp/bun-issue/index.html +85ms
transforming (1) index.html  vite:resolve-details [url] /src/main.ts -> D:/temp/bun-issue/src/main.ts +21ms
  vite:resolve-details [relative] ./style.css -> D:/temp/bun-issue/src/style.css +10ms
  vite:resolve-details [relative] ./typescript.svg -> D:/temp/bun-issue/src/typescript.svg +0ms
  vite:resolve-details [fallthrough] /vite.svg +1ms
  vite:resolve-details [relative] ./counter.ts -> D:/temp/bun-issue/src/counter.ts +0ms
  vite:resolve-details [fs] D:\temp\bun-issue\src\js\test.ts -> D:/temp/bun-issue/src/js/test.ts +1ms
  vite-tsconfig-paths resolved: {
  id: '@js/test.ts',
  importer: 'D:/temp/bun-issue/src/main.ts',
  resolvedId: 'D:/temp/bun-issue/src/js/test.ts',
  configPath: 'D:/temp/bun-issue/tsconfig.json'
} +70ms
✓ 8 modules transformed.
dist/index.html                 0.46 kB │ gzip: 0.29 kB
dist/assets/index-Cz4zGhbH.css  1.21 kB │ gzip: 0.62 kB
dist/assets/index-v3LfW65U.js   3.07 kB │ gzip: 1.65 kB
✓ built in 83ms
```

**Comment by aleclarson** [comment]
Probably fixed by #157 (available in v5.1.0). If you're still experiencing issues on Windows, please open a new issue.
